### PR TITLE
Lucene 8 10

### PR DESCRIPTION
**Pull request by sbusk**

### CODE REVIEW

**Review by jbartolini — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -74,7 +74,7 @@
         <jackson.version>2.12.5</jackson.version>
 
         <log4j.version>2.14.1</log4j.version>
-        <lucene.version>7.7.3</lucene.version>
+        <lucene.version>8.9.0</lucene.version>
```
Comment (eshryane) [inline pom.xml:77]: Add a comment above that the dependency version must be synchronised with Elasticsearch

File: whois-api/src/test/java/net/ripe/db/whois/api/fulltextsearch/FullTextAnalyzerTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * The {@link org.apache.lucene.analysis.Analyzer} used in whois does *not* split on punctuation at the moment.
+ * This has been done to support finding objects based on exact key containing interpunction.
+ * Downside of not stripping the punctuation is that the index is blown up considerably because for example we store both
+ * "members" and "members:" in the index.
+ * We probably want to implement a filter that strips most punctuation but also emits punctuation that is part of an object's key.
+ * This test has been added to more quickly find the cause of changes in fulltext search results as a result of Lucene upgrades or Analyzer changes :)
+ */
+public class FullTextAnalyzerTest {
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/fulltextsearch/FullTextAnalyzerTest.java:25]: This test class is not related to the Lucene 8 update? Remove from this PR?

Comment (sbusk) [reply whois-api/src/test/java/net/ripe/db/whois/api/fulltextsearch/FullTextAnalyzerTest.java:25]: It is something I have added as a result of losing time figuring out why the behaviour was different after a Lucene major version bump, trying to use the StandardAnalyzer instead but then figuring out that we can't and I want to have it documented and under test for the next time we do a major version upgrade and run into the same problem. So IMHO yes it is related to the Lucene 8 upgrade.

File: pom.xml (hunk 1 above)
Comment (sbusk) [reply pom.xml:77]: Good one, will do.

File: whois-api/src/test/java/net/ripe/db/whois/api/fulltextsearch/FullTextAnalyzerTest.java (hunk 2 above)
Comment (eshryane) [reply whois-api/src/test/java/net/ripe/db/whois/api/fulltextsearch/FullTextAnalyzerTest.java:25]: OK! Thanks. I confused this with Jacopo's changes for the logsearch analyser, but they are not related.

**Review by eshryane — APPROVED** [review]
